lib/codecs/src/decoding/format/avro.rs

```diff
@@ -0,0 +1,371 @@
+        AvroValue::Array(array) => {
+            let mut vector = Vec::new();
+            for item in array {
+                vector.push(try_from(item)?);
+            }
+            Ok(VrlValue::Array(vector))
+        }
+        AvroValue::Boolean(boolean) => Ok(VrlValue::from(boolean)),
+        AvroValue::Bytes(bytes) => Ok(VrlValue::from(bytes)),
+        AvroValue::Date(d) => Ok(VrlValue::from(d)),
+        AvroValue::Decimal(_) => {
+            return Err(vector_common::Error::from(format!(
+                "AvroValue::Decimal is not supported"
+            )))
+        }
+        AvroValue::Double(double) => Ok(VrlValue::from_f64_or_zero(double)),
+        AvroValue::Duration(_) => Err(
+            vector_common::Error::from(format!("AvroValue::Duration is not supported")),
+        ),
+        AvroValue::Enum(_, string) => Ok(VrlValue::from(string)),
+        AvroValue::Fixed(_, bytes) => Ok(VrlValue::from(bytes)),
+        AvroValue::Float(float) => Ok(VrlValue::from_f64_or_zero(float as f64)),
+        AvroValue::Int(int) => Ok(VrlValue::from(int)),
+        AvroValue::Long(long) => Ok(VrlValue::from(long)),
+        AvroValue::Map(items) => items
+            .into_iter()
+            .map(|(key, value)| try_from(value).map(|v| (KeyString::from(key) , v)))
+            .collect::<Result<Vec<_>, _>>()
+            .map(|v| VrlValue::Object(v.into_iter().collect())),
+        AvroValue::Null => Ok(VrlValue::Null),
+        AvroValue::Record(items) => items
+            .into_iter()
+            .map(|(key, value)| try_from(value).map(|v| (KeyString::from(key), v)))
+            .collect::<Result<Vec<_>, _>>()
+            .map(|v| VrlValue::Object(v.into_iter().collect())),
+        AvroValue::String(string) => Ok(VrlValue::from(string)),
+        AvroValue::TimeMicros(time_micros) => Ok(VrlValue::from(time_micros)),
+        AvroValue::TimeMillis(time_millis) => Ok(VrlValue::from(time_millis)),
+        AvroValue::TimestampMicros(micros) => Ok(VrlValue::Timestamp(
+            Utc.timestamp_micros(micros)
+                .single()
+                .ok_or("failed to convert TimestampMicros")?,
+        )),
+        AvroValue::TimestampMillis(millis) => Ok(VrlValue::Timestamp(
+            Utc.timestamp_millis_opt(millis)
+                .single()
+                .ok_or("failed to convert TimestampMillis")?,
+        )),
+        AvroValue::Union(_, v) => try_from(*v),
+        AvroValue::Uuid(uuid) => Ok(VrlValue::from(uuid.as_hyphenated().to_string())),
+        AvroValue::LocalTimestampMillis(_) | AvroValue::LocalTimestampMicros(_) => Err(
+            vector_common::Error::from(format!("AvroValue::LocalTimestampMillis and AvroValue::LocalTimestampMicros is not supported")),
+        ),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use apache_avro::Schema;
+    use bytes::BytesMut;
+    use uuid::Uuid;
+
+    use super::*;
+
+    #[test]
+    fn deserialize_avro() {
+        #[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
Nit 🥜: This is repeated across all tests, can you refactor it?

Reply:
Ye of course, have moved it out so they can share, had to chage one to use message instead of uuid but tests are passing fine with it